src/main/resources/com/google/api/codegen/java/settings.snip

```diff
@@ -144,21 +146,22 @@
   }
 
   private static String getGapicVersion() {
-    try {
-      Properties properties = new Properties();
-      String mavenPropertiesPath = META_FILE_ROOT
-          + GROUP_ID + "/"
-          + ARTIFACT_ID + "/pom.properties";
-      InputStream inputStream =
-          ImageAnnotatorSettings.class.getResourceAsStream(mavenPropertiesPath);
-      if (inputStream != null) {
-        properties.load(inputStream);
-        return properties.getProperty(META_VERSION_KEY, "");
+    if (gapicVersion == null) {
+      try {
+        Properties properties = new Properties();
```

Comment:
GaxPropertiesProvider is currently used for retrieving data from gax.properties. We can consider renaming it to something like `MetadataProvider` to support more general usage.